homeassistant/components/recorder/statistics.py

```diff
@@ -2066,25 +2078,44 @@ def _sorted_statistics_to_dict(
             convert = _get_statistic_to_display_unit_converter(unit, state_unit, units)
         else:
             convert = None
-        ent_results = result[statistic_id]
+        ent_results_append = result[statistic_id].append
+        #
+        # The below loop is a red hot path for energy, and every
+        # optimization counts in here.
+        #
+        # Specifically, we want to avoid function calls,
+        # attribute lookups, and dict lookups as much as possible.
+        #
         for db_state in stats_list:
             row: dict[str, Any] = {
-                "start": (start_ts := db_state.start_ts),
```

Comment:
sqlalchemy wants to avoid the memory overhead of storing rows in a dict, so the `Row` object is a wrapper that has a dict under it that has to resolve each attribute every time its called .. by doing a `dict` lookup.

The TODO at the top of https://github.com/sqlalchemy/sqlalchemy/blob/88c9463dc1114d493ed886a11705af714f27a221/lib/sqlalchemy/cyextension/resultproxy.pyx#L1 is a bit of gold